test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/MockTracingTelemetry.java

```diff
@@ -40,7 +46,12 @@ public TracingContextPropagator getContextPropagator() {
 
     @Override
     public void close() {
-        ((StrictCheckSpanProcessor) spanProcessor).ensureAllSpansAreClosed();
-        ((StrictCheckSpanProcessor) spanProcessor).clear();
+        List<MockSpanData> spanData = ((StrictCheckSpanProcessor) spanProcessor).getFinishedSpanItems();
+        if (spanData.size() != 0) {
+            TelemetryValidators validators = new TelemetryValidators(
+                Arrays.asList(AllSpansAreEndedProperly.class, AllSpansAreInOrder.class, AllSpansHaveUniqueId.class)
```

Comment:
Why are we passing validators as classes instead on the concrete objects? Also, this is the same code that is being called from the unit test cases as well?

Reply:
Changes made to pass objects